nemo/collections/asr/parts/mixins/mixins.py

```diff
@@ -112,43 +200,47 @@ def get_vocab():
                 return vocabulary
 
             # attach utility values to the tokenizer wrapper
-            self.tokenizer.tokenizer.vocab_size = len(vocabulary)
-            self.tokenizer.tokenizer.get_vocab = get_vocab
-            self.tokenizer.tokenizer.all_special_tokens = self.tokenizer.special_token_to_id
+            tokenizer.tokenizer.vocab_size = len(vocabulary)
+            tokenizer.tokenizer.get_vocab = get_vocab
+            tokenizer.tokenizer.all_special_tokens = tokenizer.special_token_to_id
 
         else:
             # This is a WPE Tokenizer
             # If path from previous registration exists, remove it
-            if 'vocab_path' in self.tokenizer_cfg:
-                vocab_path = self.tokenizer_cfg.get('vocab_path')
+            if 'vocab_path' in tokenizer_cfg:
```

Comment:
Also needs to be split as above. One func for mono, one for multi lang, and two split funcs for calling and registering the Tokenizers (as well as calling the tokenizer building funcs)

Reply:
done